helix-view/src/gutter.rs

```diff
@@ -50,6 +52,31 @@ pub fn diagnostic<'doc>(
     })
 }
 
+pub fn diff<'doc>(
+    _editor: &'doc Editor,
+    doc: &'doc Document,
+    _view: &View,
+    theme: &Theme,
+    _is_focused: bool,
+    _width: usize,
+) -> GutterFn<'doc> {
+    let added = theme.get("diff.plus");
+    let deleted = theme.get("diff.minus");
+    let modified = theme.get("diff.delta");
+
+    Box::new(move |line: usize, _selected: bool, out: &mut String| {
+        let diff = *doc.differ()?.get_line_diffs().get(&line)?;
```

Comment:
Should be able to store `doc.differ()?.get_line_diffs()` in a var right outside the closure, so that this isn't called per line (just once per render)

Reply:
good catch this could in theory even lead to temporarly inconsistent gutter (if the line diffs change during gutter rendering).